frontend/src/modules/Shared/Shares/ShareEditForm.js

```diff
@@ -221,15 +221,28 @@ export const ShareEditForm = (props) => {
   const [requestPurpose, setRequestPurpose] = useState('');
   const [smthChanged, setSmthChanged] = useState(false);
 
+  const canUpdateRequestPurpose = () => {
+    return (
+      (share.userRoleForShareObject === 'Requesters' ||
+        share.userRoleForShareObject === 'ApproversAndRequesters') &&
+      (share.status === 'Draft' ||
+        share.status === 'Processed' ||
+        share.status === 'Rejected' ||
+        share.status === 'Submitted')
+    );
+  };
+
   const handlePageChange = async (event, value) => {
     if (value <= sharedItems.pages && value !== sharedItems.page) {
       await setFilter({ ...filter, page: value });
     }
   };
 
   const beforeClose = async () => {
-    if (smthChanged || requestPurpose !== share.requestPurpose) {
+    if (requestPurpose !== share.requestPurpose) {
```

Comment:
need to a `setSmthChanged(true);` for if no response errors for `updateRequestPurpose` then

Reply:
updateRequestPurpose doesn't change the state of Share. So, basically, there is nothing to submit.

Reply:
@noah-paige may be we should change that, but not before 2.6

Reply:
`onApply` just refreshed the ShareObject and associated share items in the view

If we do not run `onApply()` then I can update the request purpose --> hit `Save & Close` --> the share view is not updated and the old request purpose still shows